tensorflow/python/training/slot_creator.py

```diff
@@ -42,18 +42,19 @@
 from tensorflow.python.framework import ops
 from tensorflow.python.ops import array_ops
 from tensorflow.python.ops import variables
+from tensorflow.python.ops import variable_scope
 
 
 def _create_slot_var(primary, val, scope):
   """Helper function for creating a slot variable."""
 
-  slot = variables.Variable(val, name=scope, trainable=False)
+  slot = variable_scope.get_variable(scope, initializer=val, trainable=False)
   # pylint: disable=protected-access
   if isinstance(primary, variables.Variable) and primary._save_slice_info:
     # Primary is a partitioned variable, so we need to also indicate that
     # the slot is a partitioned variable.  Slots have the same partitioning
     # as their primaries.
-    real_slot_name = scope[len(primary.op.name + "/"):-1]
+    real_slot_name = slot.name[len(primary.op.name + "/"):-2]
```

Comment:
Because `scope` is ended with` linear//weight/Adam/`, so remove `linear//weight/` and `/.` But for slot.name, it's ended with `linear//weight/Adam:0,` so remove `linear//weight/` and `:0.`